python/adelphi/adelphi/store.py

```diff
@@ -39,20 +40,26 @@
                     "system_views"])
 
 def build_auth_provider(username = None,password = None):
-    # instantiate auth provider if credentials have been provided
-    auth_provider = None
-    if username is not None and password is not None:
-        auth_provider = PlainTextAuthProvider(username=username, password=password)
-    return auth_provider
+    """Instantiate auth provider if credentials have been provided"""
+    if username is None or password is None:
+        return None
+    return PlainTextAuthProvider(username=username, password=password)
 
 
+@retry
```

Comment:
Annotated to make this a tenacity-aware function but we don't provide global configs here since those are largely context-specific.  Individual callers can provide args appropriate for their specific use; see the example in test-adelphi to see what this looks like.